app/components/Sidebar/components/DocumentLink.js

```diff
@@ -101,48 +108,87 @@ function DocumentLink({
 
   const [menuOpen, setMenuOpen] = React.useState(false);
 
+  // DnD
+
+  // Draggable
+  const [{ isDragging }, drag] = useDrag({
+    item: { type: "document", id: node.id },
+    collect: (monitor) => ({
+      isDragging: !!monitor.isDragging(),
+    }),
+  });
+
+  // Droppable
+  const [{ isOver, canDrop }, drop] = useDrop({
+    accept: "document",
+    drop: () => console.log("dropped on: ", node.title, { node }),
+    canDrop: (item, monitor) =>
+      pathToNode && !pathToNode.includes(monitor.getItem().id),
+    collect: (monitor) => ({
+      isOver: !!monitor.isOver(),
+      item: monitor.getItem(),
+      canDrop: monitor.canDrop(),
+    }),
+  });
+
+  // /DnD
+  // cant drag onto itself or inside its own shit
```

Comment:
lol

Reply:
That was a todo which is actually done in this commit already!

Reply:
- [ ] placeholder after every document
- [ ] placeholder before first document
- [ ] up one level (move out of nesting)
- [ ] down one level (move into nesting)